Analysis/Tasks/PWGHF/HFLcK0sPCandidateSelector.cxx

```diff
@@ -0,0 +1,400 @@
+  {
+    if (track.pt() < d_pidTPCMinPt || track.pt() >= d_pidTPCMaxPt) {
+      LOG(DEBUG) << "Bachelor pt is " << track.pt() << ", we trust TPC PID in [" << d_pidTPCMinPt << ", " << d_pidTPCMaxPt << "]";
+      return false;
+    }
+    return true;
+  }
+
+  /// Check if we will use TPC PID
+  /// \param track is the track
+  /// \note function to be expanded
+  /// \return true if track is ok for TPC PID
+  template <typename T>
+  bool applyTPCPID(const T& track)
+  {
+    if (TMath::Abs(track.pt()) < d_applyPidTPCMinPt) {
+      LOG(DEBUG) << "Bachelor pt is " << track.pt() << ", we apply TPC PID from " << d_applyPidTPCMinPt;
+      return false;
+    }
+    LOG(DEBUG) << "Bachelor pt is " << track.pt() << ", we apply TPC PID from " << d_applyPidTPCMinPt;
+    return true;
+  }
+
+  /// Check if track is ok for TOF PID
+  /// \param track is the track
+  /// \note function to be expanded
+  /// \return true if track is ok for TOF PID
+  template <typename T>
+  bool validCombPID(const T& track)
+  {
+    if (TMath::Abs(track.pt()) > d_pidCombMaxP) { // is the pt sign used for the charge? If it is always positive, we should remove the abs
+      return false;
+    }
+    return true;
+  }
+
+  /// Check if track is compatible with given TPC Nsigma cut for a given flavour hypothesis
+  /// \param track is the track
+  /// \param nPDG is the flavour hypothesis PDG number
+  /// \param nSigmaCut is the nsigma threshold to test against
+  /// \note nPDG=211 pion  nPDG=321 kaon
+  /// \return true if track satisfies TPC PID hypothesis for given Nsigma cut
+  template <typename T>
+  bool selectionPIDTPC(const T& track, double nSigmaCut)
+  {
+    double nSigma = 100.0; //arbitarily large value
+    nSigma = track.tpcNSigmaPr();
+    LOG(DEBUG) << "nSigma for bachelor = " << nSigma << ", cut at " << nSigmaCut;
+    return std::abs(nSigma) < nSigmaCut;
+  }
+
+  /*
+  /// Check if track is compatible with given TOF NSigma cut for a given flavour hypothesis
+  /// \param track is the track
+  /// \param nPDG is the flavour hypothesis PDG number
+  /// \param nSigmaCut is the nSigma threshold to test against
+  /// \note nPDG=211 pion  nPDG=321 kaon
+  /// \return true if track satisfies TOF PID hypothesis for given NSigma cut
+  template <typename T>
+  bool selectionPIDTOF(const T& track, int nPDG, double nSigmaCut)
+  {
+    double nSigma = 100.0; //arbitarily large value
+    nPDG = TMath::Abs(nPDG);
+    if (nPDG == 111) {
+      nSigma = track.tofNSigmaPi();
+    } else if (nPDG == 321) {
+      nSigma = track.tofNSigmaKa();
+    } else {
+      return false;
+    }
+    return nSigma < nSigmaCut;
+  }
+  */
+
+  /// PID selection on daughter track
+  /// \param track is the daughter track
+  /// \param nPDG is the PDG code of the flavour hypothesis
+  /// \note nPDG=211 pion  nPDG=321 kaon
+  /// \return 1 if successful PID match, 0 if successful PID rejection, -1 if no PID info
+  template <typename T>
+  int selectionPID(const T& track)
```

Comment:
All the PID selection will be replaced by a central PID selector. All this PID-related code can be removed.

Reply:
Ok, but should I remove it now?